BUG: Regression with

### Describe the bug

#10717 (cc @marxin, @AA-Turner) broke things in sphinx-gallery tests:

https://dev.azure.com/sphinx-gallery/sphinx-gallery/_build/results?buildId=881&view=logs&j=afeff7a1-1360-5c62-1669-7f018f69783f&t=b0b17f84-1475-5e62-5f43-b0e7d65ffba0&l=232

```
sphinx_gallery/tests/test_full.py:569: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../sphinx/sphinx/application.py:350: in build
    self.builder.build_update()
../sphinx/sphinx/builders/__init__.py:301: in build_update
    self.build(to_build,
../sphinx/sphinx/builders/__init__.py:367: in build
    self.write(docnames, list(updated_docnames), method)
../sphinx/sphinx/builders/__init__.py:551: in write
    self.prepare_writing(docnames)
../sphinx/sphinx/builders/html/__init__.py:470: in prepare_writing
    self.load_indexer(docnames)
../sphinx/sphinx/builders/html/__init__.py:947: in load_indexer
    self.indexer.prune(keep)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sphinx.search.IndexBuilder object at 0x7fb4081792a0>
docnames = {'auto_examples/future/plot_future_imports', 'auto_examples/local_module', 'auto_examples/plot_animation', 'auto_examples/plot_command_line_args', 'auto_examples/plot_defer_figures', 'auto_examples/plot_log', ...}

    def prune(self, docnames: Iterable[str]) -> None:
        """Remove data for all docnames not in the list."""
        new_titles = {}
        new_alltitles = {}
        new_filenames = {}
        for docname in docnames:
            if docname in self._titles:
                new_titles[docname] = self._titles[docname]
>               new_alltitles[docname] = self._all_titles[docname]
E               KeyError: 'auto_examples/local_module'

../sphinx/sphinx/search/__init__.py:399: KeyError
```

### How to Reproduce

```
$ git clone https://github.com/sphinx-gallery/sphinx-gallery.git
$ cd sphinx-gallery
$ pip install matplotlib pytest
$ pytest sphinx_gallery -k rebuild
```

### Expected behavior

No error

### Your project

https://github.com/sphinx-gallery/sphinx-gallery/

### Screenshots

_No response_

### OS

Linux

### Python version

3.10

### Sphinx version

5.0dev

### Sphinx extensions

sphinx-gallery

### Extra tools

_No response_

### Additional context

_No response_

--- a/sphinx/search/__init__.py
+++ b/sphinx/search/__init__.py
@@ -290,9 +290,11 @@ class IndexBuilder:
         self._titles = dict(zip(index2fn, frozen['titles']))
         self._all_titles = {}
 
+        for docname in self._titles.keys():
+            self._all_titles[docname] = []
         for title, doc_tuples in frozen['alltitles'].items():
             for doc, titleid in doc_tuples:
-                self._all_titles.setdefault(index2fn[doc], []).append((title, titleid))
+                self._all_titles[index2fn[doc]].append((title, titleid))
 
         def load_terms(mapping: Dict[str, Any]) -> Dict[str, Set[str]]:
             rv = {}
@@ -380,12 +382,12 @@ class IndexBuilder:
         alltitles: Dict[str, List[Tuple[int, str]]] = {}
         for docname, titlelist in self._all_titles.items():
             for title, titleid in titlelist:
-                alltitles.setdefault(title, []).append((fn2index[docname],  titleid))
+                alltitles.setdefault(title, []).append((fn2index[docname], titleid))
 
         index_entries: Dict[str, List[Tuple[int, str]]] = {}
         for docname, entries in self._index_entries.items():
             for entry, entry_id, main_entry in entries:
-                index_entries.setdefault(entry.lower(), []).append((fn2index[docname],  entry_id))
+                index_entries.setdefault(entry.lower(), []).append((fn2index[docname], entry_id))
 
         return dict(docnames=docnames, filenames=filenames, titles=titles, terms=terms,
                     objects=objects, objtypes=objtypes, objnames=objnames,

